analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantModifier.cs

```diff
@@ -204,8 +205,8 @@ private static void CheckTypeDeclarationForRedundantPartial(SyntaxNodeAnalysisCo
         {
             var typeDeclaration = (TypeDeclarationSyntax)context.Node;
             if (typeDeclaration.Modifiers.Any(SyntaxKind.PartialKeyword)
-                && context.ContainingSymbol.Kind == SymbolKind.NamedType
-                && context.SemanticModel.GetDeclaredSymbol(typeDeclaration) is { DeclaringSyntaxReferences: { Length: <= 1 } })
+                && !context.IsRedundantPositionalRecordContext()
```

Comment:
_Performance:_ This should be unmeasurably better than the `Any` collection iteration check above.